### ISSUE TITLE
URL issues

### ISSUE THREAD

**Issue by Ace2213** [ConnectedCommunities/humhub-modules-questionanswer#63]
Within spaces, the URLs seem to be broken. It's not adding /s/space-name to the URL, which results in a 404 error.

And I have yet to see the global page, possibly due to the same reason.

And "Home" in the breadcrumb leads nowhere and does nothing as far as I can tell.

**Comment by Ace2213** [comment]
Fixed the URLs by changing them in the view files from `Url::createUrl` to `$this->context->contentContainer->createUrl`

Global page still inaccessible, making the mode choice useless.

Tag view still 404s even with the links corrected.

The breadcrumb is still useless since I've not been able to use spaces as categories, so the choices are Home (which does nothing), Index which is the current view and [space-name] which is the same as Index.

The admin view is cool I guess if you have a lot of questions. The TagWallEntry... why?

Basically there's no difference between global mode and space mode because either way the questions are isolated. The index (at least with my fix) goes to /s/space-name/index or /u/-user-name/index so it only shows those questions. Without my fix, the Index 404s because there's no global page.

It would be nice if this was fixed to work as intended, because it sounds cool. In its current state, I've disabled many of its features due to their not working (tags, breadcrumb, global page, admin page)

**Comment by Ace2213** [comment]
Also, in Answer stream entry, there's a bit of code hanging off after the question. That href tag is completely messed up, but I was able to fix it.

Some URLs are still broken, in stream entries and  the onsubmit redirect URLs. I can't seem to fix those.

**Comment by benmag** [comment]
If you submit a pull request for these changes I will accept it

> Basically there's no difference between global mode and space mode because either way the questions are isolated. 

The different modes are more visual than anything. Global (when it works) should just show all the topics in one view where as spaces mode is for grouping categories of topics

**Comment by Ace2213** [comment]
Yeah and the global mode sounds great, but it never works. domain.com/questionanswer always 404s. It seems like there's no corresponding page/route. Only domain.com/s/space-name/questionanswer or domain.com/u/user-name/questionaswer are currently working

**Comment by benmag** [comment]
Have you tried accessing via `/index.php?r=questionanswer%2Fquestion%2Findex`?

**Comment by Ace2213** [comment]
I have pretty URLs enabled, so that would be domain.com/questionanswer

**Comment by Ace2213** [comment]
In its current state, with the fixes I've committed, I would still recommend disabling tags (everything related to them), the admin page, related questions and the breadcrumb. That's how I'm using it.

The only issue that still exists WITH all of these modifications and disables, is the URLs in the stream entries. I haven't been able to fix those.

Can you take a look at that? Because if you can fix that, I can commit the changes required to disable the above features and make the module fully functional, albeit slightly more limited.

**Comment by ArchBlood** [comment]
Something similar to how the Custom Pages module should be implemented.

https://github.com/humhub/humhub-modules-custom-pages/blob/master/config.php#L16

**Comment by Ace2213** [comment]
That could fix the global page, and maybe tags too. I'll give it a shot later. That would bring back 3 essential features of this module. (Global page/mode, tags and the breadcrumb)

I still don't know how to fix the URLs on stream entries, notifications and votes.
